drivers/sensor/stmemsc/stmemsc_i2c.c

```diff
@@ -19,6 +20,21 @@ int stmemsc_i2c_read(const struct stmemsc_cfg_i2c *stmemsc,
 int stmemsc_i2c_write(const struct stmemsc_cfg_i2c *stmemsc,
 			      uint8_t reg_addr, uint8_t *value, uint8_t len)
 {
+	/* NRF TWIM does not support burst write. We can't detect whether the
+	 * I2C master uses TWI or TWIM, so use a substitute implementation
+	 * unconditionally on Nordic.
+	 */
+	if (IS_ENABLED(CONFIG_I2C_NRFX)) {
```

Comment:
Since it seems that using i2c_burst_write() is deprecated (#12588) why not calling always the i2c_write() and get rid of the i2c_burst_write()? I mean, using this piece of code not only for Nordic platform but always, following the i2c API comment from @pabigot:

 * @warning The combined write synthesized by this API may not be
 * supported on all I2C devices.  Uses of this API may be made more
 * portable by replacing them with calls to i2c_write() passing a
 * buffer containing the combined address and data.

Reply:
To be honest, as ST sensor maintainer I would like to see them working properly with all platforms. And it is not very clean to have a check of the platfrom here in this code. Better get rid of it. @richardbarlow would you kindly re-implement this PR so that I could also verify it on my platforms?

Reply:
> Since it seems that using i2c_burst_write() is deprecated (#12588)

I think that is a rumor.

>why not calling always the i2c_write() and get rid of the i2c_burst_write()

Because it is generally very useful function and many operating systems offer it?